dashboard/config/levels/custom/aichat/AI Chat Test Level - 2.level

```diff
@@ -12,11 +12,46 @@
     "bot_title": "Deprecated",
     "bot_description": "Deprecated",
     "disable_projects": true,
+    "initial_ai_customizations": {
+      "botName": {
+        "value": "",
+        "visibility": "editable"
+      },
+      "temperature": {
+        "value": 0.5,
+        "visibility": "editable"
+      },
+      "systemPrompt": {
+        "value": "",
+        "visibility": "editable"
+      },
+      "retrievalContexts": {
+        "value": [
+
+        ],
+        "visibility": "editable"
+      },
+      "modelCardInfo": {
+        "value": {
+          "description": "",
+          "intendedUse": "",
+          "limitationsAndWarnings": "",
+          "testingAndEvaluation": "",
+          "exampleTopics": [
+
+          ]
+        },
+        "visibility": "editable"
+      },
+      "hidePresentationPanel": true
+    },
     "aichat_settings": {
       "initialCustomizations": {
+        "selectedModelId": "chatgpt",
```

Comment:
Do we handle levels that don't have this configuration in them, or do we need to update them all?

Separately, notice that we're writing the old style of initial customizations above:

```
      "botName": {
        "value": "",
        "visibility": "editable"
      },
      "temperature": {
        "value": 0.5,
        "visibility": "editable"
      },
```

But I get that too when I try to make an edit (guessing it's bc we have the old config in our database entries for these levels). So maybe fine for now.

Reply:
If I manually clean file, run `bundle exec rake seed:custom_levels`, then edit the file via levelbuilder UI, the old config didn't reappear for me.
Either way, we should probably clean up the old config.`

Reply:
> Do we handle levels that don't have this configuration in them, or do we need to update them all?

I believe that the following handles:
https://github.com/code-dot-org/code-dot-org/blob/53f6f88e0de265636288548606ca077a3969d41d/apps/src/aichat/views/modelCustomization/SetupCustomization.tsx#L42-L47

There is a weird case for an existing .level file that if these are not defined, then the compare models button doesn't appear. But for new `aichat` levels, I don't think this should be an issue.

code-dot-org/apps/src/aichat/views/modelCustomization/SetupCustomization.tsx

Lines 42 to 47 in 53f6f88

	 const availableModels = availableModelIds 
	   ? modelDescriptions.filter(model => availableModelIds.includes(model.id)) 
	   : modelDescriptions; 
	  
	 const chosenModelId = 
	   aiCustomizations.selectedModelId || availableModels[0].id; 
There is a weird case for an existing .level file that if these are not defined, then the compare models button doesn't appear. But for new aichat levels, this shouldn't be an issue.

Reply:
I think they should work with the defaulting behavior, but I'm not 100% confident there aren't cases where the state of levels might be in weird states on different folks' machines? 

Thanks @fisher-alice for answering this and the extra info!! I want to understand better the process you're going through to manually clean the file and re-seed and stuff. Will we need to go and do something similar in each environment now, if we're done updating the level config? Could you walk me through what you did? Thanks!!

Reply:
> There is a weird case for an existing .level file that if these are not defined, then the compare models button doesn't appear. But for new aichat levels, I don't think this should be an issue.

> I think they should work with the defaulting behavior, but I'm not 100% confident there aren't cases where the state of levels might be in weird states on different folks' machines?

Yeah I'm not too worried about breaking people's local setups, but I know Dan made a level progression this week, so I think it'd be worth figuring out if this is going to break it (ie, hide compare models button)? It probably wouldn't be too hard to just manually update the level config on that handful of files.

Reply:
To test Dan's level progression, lmk if I have this right- 
I should go find the .level files from the content scoops, grab the files and put them in my local, and then seed those scripts into my db, and try to run them?

Reply:
er, I can run seed:all and it will seed everything that exists in the repo to my local db?

Reply:
>er, I can run seed:all and it will seed everything that exists in the repo to my local db which will include the new level progression, as long as I've merged in the latest staging that has the new levels??

Yes - that would do it! Thanks for checking this!

Reply:
Yeah just to +1, we should make sure to reseed our DBs before making level update changes as we're actively iterating on the level properties structure (shouldn't be as much of an issue once the structure is finalized)